### ISSUE TITLE
With complex nesting, wrong indentation isn't caught for all items

### ISSUE THREAD

**Issue by bcbrookman** [adrienverge/yamllint#485]
In the file below, I'd expect the indentation for all three "foo" maps nested in the sequences under the "items" to be caught by the default "wrong indentation" rules.

```
---
- item1:
  - foo:
      bar: baz
  item2:
  - foo:
      bar: baz
  item3:
  - foo:
      bar: baz

```
However, the first one doesn't seem to be caught; only the subsequent ones are.

```shell
$ yamllint -d default test.yaml 
test.yaml
  6:3       error    wrong indentation: expected 4 but found 2  (indentation)
  9:3       error    wrong indentation: expected 4 but found 2  (indentation)

```

If I properly indent only the last two as shown below, I get no errors.

```
---
- item1:
  - foo:
      bar: baz
  item2:
    - foo:
        bar: baz
  item3:
    - foo:
        bar: baz

```
The provided examples, of course, look unnecessarily complicated, but I uncovered this issue while linting an [Ansible playbook with handlers](https://docs.ansible.com/ansible/latest/user_guide/playbooks_handlers.html#handler-example). I'm unable to change the layout of the file. Regardless, I feel that yamllint should detect this wrong indentation, in general.

I was using yamllint 1.26.3 installed via pip, and also upgraded to 1.27.1 with the same behavior.

I also didn't find any other issues reported that seem to match what I've encountered. Thanks!

**Comment by adrienverge** [comment]
Hello @bcbrookman, first of all thanks for this comprehensive and precise report!

That's a very interesting case, I wasn't aware of this bug. I noticed that it's not about "the first one", it occurs for *all* keys, until a sub-item (`bar: baz`) is inserted:
```yaml
- item1:
  - foo:
  item2:
  - foo:
  item3:
  - foo:
      bar: baz
  item4:
  - foo:   ← the problem is reported here only
```

I guess something needs to be fixed in `yamllint/rules/indentation.py`, and a new test case added. Contributions are welcome!

**Comment by bcbrookman** [comment]
I dug into this and found that it's related to the arbitrary expected indent value of 2 is set when a block sequence should be indented, but the expected indent value is unknown/hasn't been computed yet. I've proposed PR #491 as a potential fix.

**Comment by erlangparasu** [comment]
any updates?

**Comment by bcbrookman** [comment]
Hi @erlangparasu, this was fixed in #491 but the issue was left open. Unless there are objections, I think we can close it at this point (nearly two years later 😆)

**Comment by adrienverge** [comment]
I confirm this is fixed since yamllint 1.29.0 :+1:

**Comment by erlangparasu** [comment]
oh thanks
